### ISSUE TITLE
[BITV/A11Y] `NcAppNavigationNewItem` should be implemented as button not span.

### ISSUE THREAD

**Issue by Fenn-CS** [nextcloud/nextcloud-vue#4108]
Following [nextcloud/server #37135](https://github.com/nextcloud/server/issues/37135) it appears the right `NcAppNavigationItem` to use would be `NcAppNavigationNewItem`. That's better than using an anchor tag as it was with `NcAppNavigationItem`. However, `NcAppNavigationNewItem` by virtue of how it functions triggers and action without routing anywhere so it should be semantically implemented as a button. 

Code : https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/NcAppNavigationNewItem/NcAppNavigationNewItem.vue

https://github.com/nextcloud/nextcloud-vue/blob/2d6245af80f9cfc5004d67489d00a4b22774e350/src/components/NcAppNavigationNewItem/NcAppNavigationNewItem.vue#L79-L81

_Helpful Notes : https://karlgroves.com/links-are-not-buttons-neither-are-divs-and-spans/_
